From c4af56a39cf113715dc508907724811e5d5ea71d Mon Sep 17 00:00:00 2001
From: Marcel Bargull <marcel.bargull@udo.edu>
Date: Sat, 3 Dec 2022 20:38:08 +0100
Subject: [PATCH] Fix: Remove invalid memoization for MetaData

---
 conda_build/environ.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/conda_build/environ.py b/conda_build/environ.py
index 348b0bcd..5a57359a 100644
--- a/conda_build/environ.py
+++ b/conda_build/environ.py
@@ -663,7 +663,6 @@ def system_vars(env_dict, m, prefix):
     return os_vars(m, prefix)
 
 
-@lru_cache(maxsize=None)
 def os_vars(m, prefix):
     d = dict()
     # note the dictionary is passed in here - variables are set in that dict if they are non-null
