From 93847ecf44513cb5efba7192b179489ff458cca0 Mon Sep 17 00:00:00 2001
From: Jannis Leidel <jannis@leidel.info>
Date: Tue, 23 Nov 2021 15:20:09 +0100
Subject: [PATCH 1/3] Fix import for auxlib.

Fix 4333.
---
 conda_build/environ.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/conda_build/environ.py b/conda_build/environ.py
index ca177c5d61..ff12add8a7 100644
--- a/conda_build/environ.py
+++ b/conda_build/environ.py
@@ -188,7 +188,10 @@ def get_git_info(git_exe, repo, debug):
         parts = output.rsplit('-', 2)
         if len(parts) == 3:
             d.update(dict(zip(keys, parts)))
-        from conda._vendor.auxlib.packaging import _get_version_from_git_tag
+        try:
+            from conda._vendor.auxlib.packaging import _get_version_from_git_tag
+        except ImportError:
+            from conda.auxlib.packaging import _get_version_from_git_tag
         d['GIT_DESCRIBE_TAG_PEP440'] = str(_get_version_from_git_tag(output))
     except subprocess.CalledProcessError:
         msg = (

From bcf538bd519727f46f90ae56a5ec93f6e870cc1b Mon Sep 17 00:00:00 2001
From: Jannis Leidel <jannis@leidel.info>
Date: Tue, 23 Nov 2021 15:36:42 +0100
Subject: [PATCH 2/3] Move to conda_interface.

---
 conda_build/conda_interface.py | 7 +++++++
 conda_build/environ.py         | 7 ++-----
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/conda_build/conda_interface.py b/conda_build/conda_interface.py
index 6eb3a2fd88..444986fd70 100644
--- a/conda_build/conda_interface.py
+++ b/conda_build/conda_interface.py
@@ -31,6 +31,7 @@ def try_exports(module, attr):
 conda_46 = parse_version(CONDA_VERSION) >= parse_version("4.6.0a0")
 conda_47 = parse_version(CONDA_VERSION) >= parse_version("4.7.0a0")
 conda_48 = parse_version(CONDA_VERSION) >= parse_version("4.8.0a0")
+conda_411 = parse_version(CONDA_VERSION) >= parse_version("4.11.0a0")
 
 if conda_44:
     from conda.exports import display_actions, execute_actions, execute_plan, install_actions
@@ -47,6 +48,12 @@ def try_exports(module, attr):
     from conda.toposort import _toposort
 _toposort = _toposort
 
+if conda_411:
+    from conda.auxlib.packaging import _get_version_from_git_tag
+else:
+    from conda._vendor.auxlib.packaging import _get_version_from_git_tag
+get_version_from_git_tag = _get_version_from_git_tag
+
 from conda.exports import TmpDownload, download, handle_proxy_407  # NOQA
 from conda.exports import untracked, walk_prefix  # NOQA
 from conda.exports import MatchSpec, NoPackagesFound, Resolve, Unsatisfiable, normalized_version  # NOQA
diff --git a/conda_build/environ.py b/conda_build/environ.py
index ff12add8a7..fe19926f33 100644
--- a/conda_build/environ.py
+++ b/conda_build/environ.py
@@ -20,6 +20,7 @@
 from .conda_interface import package_cache, TemporaryDirectory
 from .conda_interface import pkgs_dirs, root_dir, create_default_packages
 from .conda_interface import reset_context
+from .conda_interface import get_version_from_git_tag
 
 from conda_build import utils
 from conda_build.exceptions import BuildLockError, DependencyNeedsBuildingError
@@ -188,11 +189,7 @@ def get_git_info(git_exe, repo, debug):
         parts = output.rsplit('-', 2)
         if len(parts) == 3:
             d.update(dict(zip(keys, parts)))
-        try:
-            from conda._vendor.auxlib.packaging import _get_version_from_git_tag
-        except ImportError:
-            from conda.auxlib.packaging import _get_version_from_git_tag
-        d['GIT_DESCRIBE_TAG_PEP440'] = str(_get_version_from_git_tag(output))
+        d['GIT_DESCRIBE_TAG_PEP440'] = str(get_version_from_git_tag(output))
     except subprocess.CalledProcessError:
         msg = (
             "Failed to obtain git tag information.\n"

From e877d00147bc771fddafb6fdf249fa2b9eb5b102 Mon Sep 17 00:00:00 2001
From: Jannis Leidel <jannis@leidel.info>
Date: Tue, 23 Nov 2021 16:02:46 +0100
Subject: [PATCH 3/3] Add news item.

---
 news/conda-411-auxlib.rst | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)
 create mode 100644 news/conda-411-auxlib.rst

diff --git a/news/conda-411-auxlib.rst b/news/conda-411-auxlib.rst
new file mode 100644
index 0000000000..8a5a9b64a5
--- /dev/null
+++ b/news/conda-411-auxlib.rst
@@ -0,0 +1,24 @@
+Enhancements:
+-------------
+
+* <news item>
+
+Bug fixes:
+----------
+
+* Handle an import from the vendored auxlib library in Conda 4.11.0 better.
+
+Deprecations:
+-------------
+
+* <news item>
+
+Docs:
+-----
+
+* <news item>
+
+Other:
+------
+
+* <news item>
