From 75f7aa6edb8407f21f189fb37f8468ee5c1f3ade Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Thu, 5 Mar 2020 22:30:30 +0100
Subject: [PATCH] Use patchelf by default

---
 conda_build/post.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conda_build/post.py b/conda_build/post.py
index 0714701a2..7c4051c29 100644
--- a/conda_build/post.py
+++ b/conda_build/post.py
@@ -544,7 +544,7 @@ def mk_relative_linux(f, prefix, rpaths=('lib',), method=None):
     rpath = ':'.join(new)
 
     # check_binary_patchers(elf, prefix, rpath)
-    if not method or not patchelf or method.upper() == 'LIEF':
+    if not patchelf or (method and method.upper() == 'LIEF'):
         set_rpath(old_matching='*', new_rpath=rpath, file=elf)
     else:
         call([patchelf, '--force-rpath', '--set-rpath', rpath, elf])
