From ec4dcc4546f9b9a3e1e61a19e2c5e89923da82c2 Mon Sep 17 00:00:00 2001
From: Ken Odegard <kodegard@anaconda.com>
Date: Thu, 26 May 2022 21:06:12 -0500
Subject: [PATCH 2/2] Fix for conda 4.13.0 Python 2.7 removal

---
 conda_build/utils.py | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/conda_build/utils.py b/conda_build/utils.py
index 6dede74a..e75812f9 100644
--- a/conda_build/utils.py
+++ b/conda_build/utils.py
@@ -2080,10 +2080,11 @@ def linked_data_no_multichannels(prefix):
     """
     from conda.core.prefix_data import PrefixData
     from conda.models.dist import Dist
-    pd = PrefixData(prefix)
-    from conda.common.compat import itervalues
-    return {Dist.from_string(prefix_record.fn, channel_override=prefix_record.channel.name):
-                prefix_record for prefix_record in itervalues(pd._prefix_records)}
+
+    return {
+        Dist.from_string(prec.fn, channel_override=prec.channel.name): prec
+        for prec in PrefixData(prefix)._prefix_records.values()
+    }
 
 
 def shutil_move_more_retrying(src, dest, debug_name):
-- 
2.35.3.windows.1

