From 74c48facd64e9c682e9387b186ef0b08a9b52548 Mon Sep 17 00:00:00 2001
From: Finn Womack <finnwomack@microsoft.com>
Date: Thu, 20 Apr 2023 13:58:00 -0700
Subject: [PATCH 1/8] add warning and return empty string

---
 conda_build/windows.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/conda_build/windows.py b/conda_build/windows.py
index 84da4a0f..af6c6b2b 100644
--- a/conda_build/windows.py
+++ b/conda_build/windows.py
@@ -110,6 +110,11 @@ def msvc_env_cmd(bits, config, override=None):
         "If this recipe does not use a compiler, this message is safe to ignore.  "
         "Otherwise, use {{compiler('<language>')}} jinja2 in requirements/build."
     )
+    if bit == "arm64":
+        log.warn(
+            "The legacy MSVC compiler setup does not support arm64 builds. "
+        )
+        return ""
     if override:
         log.warn(
             "msvc_compiler key in meta.yaml is deprecated. Use the new"
-- 
2.38.1


From e04874b64fbe7392d0c6250b5204ab6fb42c21d1 Mon Sep 17 00:00:00 2001
From: Finn Womack <flan313@gmail.com>
Date: Tue, 25 Apr 2023 15:25:47 -0700
Subject: [PATCH 2/8] Update windows.py

bit -> bits
---
 conda_build/windows.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conda_build/windows.py b/conda_build/windows.py
index af6c6b2b..f1b2eb2f 100644
--- a/conda_build/windows.py
+++ b/conda_build/windows.py
@@ -110,7 +110,7 @@ def msvc_env_cmd(bits, config, override=None):
         "If this recipe does not use a compiler, this message is safe to ignore.  "
         "Otherwise, use {{compiler('<language>')}} jinja2 in requirements/build."
     )
-    if bit == "arm64":
+    if bits == "arm64":
         log.warn(
             "The legacy MSVC compiler setup does not support arm64 builds. "
         )
-- 
2.38.1


From e249ed9a2cb136482c1ecb408a25a78b5a38e1e3 Mon Sep 17 00:00:00 2001
From: Finn Womack <flan313@gmail.com>
Date: Thu, 4 May 2023 12:32:20 -0700
Subject: [PATCH 3/8] Create news file

---
 news/made-msvc_env_cmd-no-op-for-arm64 | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)
 create mode 100644 news/made-msvc_env_cmd-no-op-for-arm64

diff --git a/news/made-msvc_env_cmd-no-op-for-arm64 b/news/made-msvc_env_cmd-no-op-for-arm64
new file mode 100644
index 00000000..a20af0a2
--- /dev/null
+++ b/news/made-msvc_env_cmd-no-op-for-arm64
@@ -0,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* <news item>
+
+### Deprecations
+
+* <news item>
+
+### Docs
+
+* <news item>
+
+### Other
+
+* Added a check to print an additional warning and return an empty string when bits is "arm64" in msvc_env_cmd
-- 
2.38.1


From 679525e439278ced5e978113abf90d975ef317db Mon Sep 17 00:00:00 2001
From: Finn Womack <flan313@gmail.com>
Date: Thu, 4 May 2023 12:33:22 -0700
Subject: [PATCH 4/8] Changed news file name

---
 ..._env_cmd-no-op-for-arm64 => made-msvc-env-cmd-no-op-for-arm64} | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 rename news/{made-msvc_env_cmd-no-op-for-arm64 => made-msvc-env-cmd-no-op-for-arm64} (100%)

diff --git a/news/made-msvc_env_cmd-no-op-for-arm64 b/news/made-msvc-env-cmd-no-op-for-arm64
similarity index 100%
rename from news/made-msvc_env_cmd-no-op-for-arm64
rename to news/made-msvc-env-cmd-no-op-for-arm64
-- 
2.38.1


From 0a91c6b586857edac7d3cd5e3ab21de1789d83f9 Mon Sep 17 00:00:00 2001
From: Finn Womack <flan313@gmail.com>
Date: Thu, 4 May 2023 12:34:16 -0700
Subject: [PATCH 5/8] Rename made-msvc-env-cmd-no-op-for-arm64 to
 arm64-msvc-env-cmd

---
 news/{made-msvc-env-cmd-no-op-for-arm64 => arm64-msvc-env-cmd} | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 rename news/{made-msvc-env-cmd-no-op-for-arm64 => arm64-msvc-env-cmd} (100%)

diff --git a/news/made-msvc-env-cmd-no-op-for-arm64 b/news/arm64-msvc-env-cmd
similarity index 100%
rename from news/made-msvc-env-cmd-no-op-for-arm64
rename to news/arm64-msvc-env-cmd
-- 
2.38.1


From f47ebd528f00a421095cb349348d36f05facaf85 Mon Sep 17 00:00:00 2001
From: Finn Womack <flan313@gmail.com>
Date: Thu, 4 May 2023 12:34:54 -0700
Subject: [PATCH 6/8] Rename arm64-msvc-env-cmd to arm64-msvc-env-cmd-no-op

---
 news/{arm64-msvc-env-cmd => arm64-msvc-env-cmd-no-op} | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 rename news/{arm64-msvc-env-cmd => arm64-msvc-env-cmd-no-op} (100%)

diff --git a/news/arm64-msvc-env-cmd b/news/arm64-msvc-env-cmd-no-op
similarity index 100%
rename from news/arm64-msvc-env-cmd
rename to news/arm64-msvc-env-cmd-no-op
-- 
2.38.1


From 477a7b7d79a774a43fcfc12ace5cf4236f2c6b15 Mon Sep 17 00:00:00 2001
From: Finn Womack <flan313@gmail.com>
Date: Thu, 4 May 2023 13:13:38 -0700
Subject: [PATCH 7/8] Rename arm64-msvc-env-cmd-no-op to
 4867-arm64-msvc-env-cmd-no-op

---
 news/{arm64-msvc-env-cmd-no-op => 4867-arm64-msvc-env-cmd-no-op} | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 rename news/{arm64-msvc-env-cmd-no-op => 4867-arm64-msvc-env-cmd-no-op} (100%)

diff --git a/news/arm64-msvc-env-cmd-no-op b/news/4867-arm64-msvc-env-cmd-no-op
similarity index 100%
rename from news/arm64-msvc-env-cmd-no-op
rename to news/4867-arm64-msvc-env-cmd-no-op
-- 
2.38.1


From ea9594bb5476b320a8fb26093a6d7452bcd046e8 Mon Sep 17 00:00:00 2001
From: "pre-commit-ci[bot]"
 <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Date: Tue, 9 May 2023 11:00:52 +0000
Subject: [PATCH 8/8] [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
---
 conda_build/windows.py | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/conda_build/windows.py b/conda_build/windows.py
index f1b2eb2f..1639c554 100644
--- a/conda_build/windows.py
+++ b/conda_build/windows.py
@@ -111,9 +111,7 @@ def msvc_env_cmd(bits, config, override=None):
         "Otherwise, use {{compiler('<language>')}} jinja2 in requirements/build."
     )
     if bits == "arm64":
-        log.warn(
-            "The legacy MSVC compiler setup does not support arm64 builds. "
-        )
+        log.warn("The legacy MSVC compiler setup does not support arm64 builds. ")
         return ""
     if override:
         log.warn(
-- 
2.38.1

